src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -4608,11 +4691,13 @@ class StubGenerator: public StubCodeGenerator {
   }
 
  public:
-  StubGenerator(CodeBuffer* code, bool all) : StubCodeGenerator(code) {
-    if (all) {
-      generate_all();
-    } else {
+  StubGenerator(CodeBuffer* code, int phase) : StubCodeGenerator(code) {
+    if (phase == 0) {
       generate_initial();
+    } else if (phase == 1) {
+      generate_phase1(); // stubs that must be available for the interpreter
+    } else {
+      generate_all();
     }
```

Comment:
Does this fix this assert?
        # guarantee(_table_length < _table_max_length) failed: Incorrect UnsafeCopyMemory::_table_max_length
I tried to trace through source code to see why and failed to do so.

Reply:
Yes it does. I've just copied what was needed from the ppc loom port and it worked. Havn't investigated further why it did.